### PR TITLE
19913 re adding parent relationship after removal throws an error

### LINKED ISSUE #19913: [code] : Re-adding parent relationship after removal throws an error
Reported via Support: [103454](https://dotcms.zendesk.com/agent/tickets/103454)
Reproducible as of dotCMS 5.2.8.3, and latest master.

The parent side of a Relationship between two Content Types can be deleted for any reason. Now, if the user decides to bring it back, an error is being displayed on both the UI and the log.

**To Reproduce**
1. Create two test Content Types named `Parent CT`, and `Child CT`
2. Go to the `Parent CT`, and add a **Relationships Field** with the following values:
2.1. Name = `Child Rel`
2.2. `New`
2.3. Content Type = `Child CT`
2.4. Relationship = `Many to many`
2.5. `Save`
3. Go to the `Child CT` this time, and add a **Relationships Field** with the following values:
3.1. Name = `Parent Rel`
3.2. `Existing`
3.3. Relationship = `ParentCt.childRel.Mano to many`
3.4. `Save`
4. Go back to the `Parent CT`, and DELETE the `Child Rel` field.
5. Drag-and-drop a new Relationships field again to bring the previous relationship back, using the following values:
5.1. Name = `Child Rel`
5.2. `Existing`
5.3. Relationship = `ParentCt.childRel.Mano to many`
5.4. `Save`
6. The following error shows up in the UI:
```
Bad request error

Unfortunately there was an error. Please try again, and if the error persists, please contact support.
```
And this error is logged to the `dotcms.log` file:
```
Caused by: com.dotcms.contenttype.exception.NotFoundInDbException: Field variable with var:parentRel not found
at com.dotcms.contenttype.business.FieldFactoryImpl.byContentTypeFieldVar(FieldFactoryImpl.java:64) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.contenttype.business.FieldAPIImpl.byContentTypeAndVar_aroundBody14(FieldAPIImpl.java:733) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.contenttype.business.FieldAPIImpl$AjcClosure15.run(FieldAPIImpl.java:1) ~[dotcms_21.01_999999.jar:?]
at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.contenttype.business.FieldAPIImpl.byContentTypeAndVar(FieldAPIImpl.java:732) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.contenttype.business.FieldAPIImpl.updateRelationshipObject(FieldAPIImpl.java:481) ~[dotcms_21.01_999999.jar:?]
at com.dotcms.contenttype.business.FieldAPIImpl.getRelationshipForField(FieldAPIImpl.java:385) ~[dotcms_21.01_999999.jar:?]
... 117 more
```

**Expected behavior**
The application needs to allow users to re-add the parent side of a Relationship after its deletion.

### PR DESCRIPTION
**Pull request by dsilvam**
* Handle exception when the relationship field is not found, which is a possible and expected scenario and continue. 
* Let the tests run without an IF which seems to be not needed
* Integration test added for described case

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java [hunk 1]
```diff
@@ -1528,6 +1528,69 @@ public void test_SaveNewRelationshipField_ShouldAddESMapping()
         }
     }
 
+    /**
+     * Method to test: com.dotcms.contenttype.business.FieldAPI#save(com.dotcms.contenttype.model.field.Field, com.liferay.portal.model.User, boolean)
+     * Given scenario: Recreate a deleted relationship field but choosing exiting relationship, from parent to child
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java:1533]: "existing" typo

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java [hunk 2]
```diff
@@ -1528,6 +1528,69 @@ public void test_SaveNewRelationshipField_ShouldAddESMapping()
...
+            assertEquals(childTypeRelationshipField.variable(), relationship.getParentRelationName());
+            assertEquals(CARDINALITY, Integer.toString(relationship.getCardinality()));
+            assertEquals(fullFieldVar, relationship.getRelationTypeValue());
+
+            final String parentTypeRelationshipFieldVariable = parentTypeRelationshipField.variable();
+            // let's delete the relationship field from the parent content type
+            APILocator.getContentTypeFieldAPI().delete(parentTypeRelationshipField);
+
+            // let's attempt to re-add the field but choosing the existing relationship
+            final Field recreatedParentTypeRelationshipField = createAndSaveRelationshipField("newRel",
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java:1576]: Before adding back the relationship again, I would validate that the relationships side is null

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java:1579]: Done

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java]: Done

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Pull request merged**